src/ripple/app/misc/Validations.cpp

```diff
@@ -104,21 +104,21 @@ class ValidationsImp : public Validations
         }
 
         auto hash = val->getLedgerHash ();
-        auto node = val->getNodeID ();
 
-        if (val->isTrusted () && isCurrent)
+        if (isCurrent &&
+            (val->isTrusted () || app_.validators ().listed (signer)))
         {
             ScopedLockType sl (mLock);
 
-            if (!findCreateSet (hash)->insert (std::make_pair (node, val)).second)
+            if (!findCreateSet (hash)->insert (std::make_pair (signer, val)).second)
```

Comment:
I think this change accounts for that scenario:
https://github.com/ripple/rippled/pull/1842/commits/1fcd583d8e269adb1d188d64b4298c3d50e7a3fc
Validations are now stored in `ValidationSet` by their master public key (when there is one).
A validation with the latest signing key replaces a validation for the same ledger sequence with a revoked signing key.